src/PowerShellEditorServices/Utility/PsesLogger.cs

```diff
@@ -0,0 +1,135 @@
+                        logLevelName, callerSourceFile, callerName, callerLineNumber, indentedLogMsg);
+                    return;
+                case LogLevel.Normal:
+                    _logger.Information("[{LogLevelName:l}] {CallerSourceFile:l}: In method '{CallerName:l}', line {CallerLineNumber}:\n{IndentedLogMsg:l}\n",
+                        logLevelName, callerSourceFile, callerName, callerLineNumber, indentedLogMsg);
+                    return;
+                case LogLevel.Warning:
+                    _logger.Warning("[{LogLevelName:l}] {CallerSourceFile:l}: In method '{CallerName:l}', line {CallerLineNumber}:\n{IndentedLogMsg:l}\n",
+                        logLevelName, callerSourceFile, callerName, callerLineNumber, indentedLogMsg);
+                    return;
+                case LogLevel.Error:
+                    _logger.Error("[{LogLevelName:l}] {CallerSourceFile:l}: In method '{CallerName:l}', line {CallerLineNumber}:\n{IndentedLogMsg:l}\n",
+                        logLevelName, callerSourceFile, callerName, callerLineNumber, indentedLogMsg);
+                    return;
+            }
+        }
+
+        /// <summary>
+        /// Log an exception in the logs.
+        /// </summary>
+        /// <param name="errorMessage">The error message of the exception to be logged.</param>
+        /// <param name="exception">The exception itself that has been thrown.</param>
+        /// <param name="callerName">The name of the method in which the ILogger is being called.</param>
+        /// <param name="callerSourceFile">The name of the source file in which the ILogger is being called.</param>
+        /// <param name="callerLineNumber">The line number in the file where the ILogger is being called.</param>
+        public void WriteException(
+            string errorMessage,
+            Exception exception,
+            [CallerMemberName] string callerName = null,
+            [CallerFilePath] string callerSourceFile = null,
+            [CallerLineNumber] int callerLineNumber = 0)
+        {
+            _logger.Error("[{Error:l}] {CallerSourceFile:l}: In method '{CallerName:l}', line {CallerLineNumber}:\n    {ErrorMessage:l}\n    {Exception:l}\n",
+                LogLevel.Error.ToString().ToUpper(), callerSourceFile, callerName, callerLineNumber, errorMessage, exception);
+        }
+
+        /// <summary>
+        /// Utility function to indent a log message by one level.
+        /// </summary>
+        /// <param name="logMessage">The log message to indent.</param>
+        /// <returns>The indented log message string.</returns>
+        private static string IndentMsg(string logMessage)
+        {
+            string[] msgLines = logMessage.Split('\n');
+
+            for (int i = 0; i < msgLines.Length; i++)
+            {
+                msgLines[i] = msgLines[i].Insert(0, "    ");
+            }
+
+            return String.Join("\n", msgLines)+"\n";
```

Comment:
Is it feasible to use `StringBuilder` here to reduce some string allocations?  Something like

```csharp
var sb = new System.Text.StringBuilder(logMessage);
return sb.Replace("\n", "\n    ").ToString();
```